pkg/nodes/queue.go

```diff
@@ -0,0 +1,108 @@
+package nodes
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/celestiaorg/torch/config"
+	"github.com/celestiaorg/torch/pkg/db/redis"
+	"github.com/celestiaorg/torch/pkg/metrics"
+)
+
+var (
+	taskQueue     = make(chan config.Peer) // taskQueue channel for pending tasks (peers to process later).
+	MaxRetryCount = 5                      // MaxRetryCount number of retries per node.
+)
+
+// ProcessTaskQueue processes the pending tasks in the queue every 5 seconds.
+func ProcessTaskQueue() {
+	ticker := time.NewTicker(5 * time.Second) // Set the interval to 5 seconds
+	//defer ticker.Stop()                       // Stop the ticker when the function exits
+
+	for {
+		select {
+		case <-ticker.C:
+			processQueue()
+		}
+	}
+}
+
+// processQueue process the nodes in the queue and tries to generate the Multi Address
+func processQueue() {
```

Comment:
I highly recommend you return errors up the call stack as the never ending loop of calls based on ticker and no exit is bad design pick